Cms/ProfileBioEdit.ascx.cs

```diff
@@ -0,0 +1,2394 @@
+
+            var personAttributeGuidList = GetAttributeValue( AttributeKey.PersonAttributesAdults ).SplitDelimitedValues().AsGuidList();
+            avcPersonAttributes.IncludedAttributes = personAttributeGuidList.Select( a => AttributeCache.Get( a ) ).ToArray();
+            avcPersonAttributes.ShowCategoryLabel = false;
+            avcPersonAttributes.NumberOfColumns = 1;
+            avcPersonAttributes.AddEditControls( person, true );
+
+            if ( rblRole.SelectedValueAsInt() == childRoleId )
+            {
+                avcPersonAttributes.Visible = false;
+                avcChildPersonAttributes.Visible = true;
+            }
+            else
+            {
+                avcPersonAttributes.Visible = true;
+                avcChildPersonAttributes.Visible = false;
+            }
+
+            // End Setup Specific Control Settings
+
+
+            // Add Controls to Panel in order in rows and columns
+
+            var pnlBody = pnlPerson.FindControl( "pnlPersonBody" );
+            var pnlFields = new Panel { ID = $"pnlFields_{person.Id}", CssClass = "d-flex flex-wrap" };
+
+            if ( pnlBody == null )
+            {
+                pnlFields = pnlPerson;
+            }
+            else
+            {
+                pnlBody.Controls.Add( pnlFields );
+            }
+
+            var visibleControlCount = 0;
+            foreach ( var ctrl in personFieldsOrder )
+            {
+                var actualCtrls = personFieldControls.Where( f => f.ID.EndsWith( ctrl ) );
+                foreach ( var actualCtrl in actualCtrls )
+                {
+                    var actualWebCtrl = actualCtrl as WebControl;
+                    if ( actualCtrl != null )
+                    {
+                        var personFieldCol = new Panel { CssClass = actualCtrl.Visible ? "col-md-6" : "" };
+                        if ( actualCtrl.Visible )
+                        {
+                            visibleControlCount++;
+                        }
+                        if ( actualWebCtrl != null && !actualWebCtrl.CssClass.Contains( "hide" ) )
+                        {
+                            pnlFields.Controls.Add( personFieldCol );
+                            personFieldCol.Controls.Add( actualCtrl );
+                        }
+                        else
+                        {
+                            visibleControlCount--;
+                            pnlFields.Controls.Add( actualCtrl );
+                        }
+                    }
+                }
+            }
+            if ( ddlGrade.Visible )
+            {
+                ScriptManager.RegisterStartupScript( ddlGrade, ddlGrade.GetType(), "grade-selection-" + BlockId.ToString() + "-" + ddlGrade.ClientID, ddlGrade.GetJavascriptForYearPicker( ypGraduation ), true );
+            }
+
+            if ( pnlFields != null && visibleControlCount < 1 )
+            {
+                pnlPerson.Visible = false;
```

Comment:
Test this scenario to verify this code does what you expect it to.

Reply:
I have tested a few different scenarios and it seems it is working as expected.